tests/accessor/accessor_implicit_conversions.h

```diff
@@ -0,0 +1,394 @@
+    }
+  }
+
+ public:
+  void operator()(const std::string& type_name,
+                  const std::string& target_name) const {
+    const auto make_section_name = [&](sycl::access_mode SrcAccessMode,
+                                       sycl::access_mode DstAccessMode,
+                                       const std::string& details) {
+      return detail::section_name_prototype<AccType, Dimension>::get(type_name)
+          .with("target", Target)
+          .with("source access", SrcAccessMode)
+          .with("destination access", DstAccessMode)
+          .with("case", details)
+          .create();
+    };
+
+    // From read-only accessor to read-only accessor
+    {
+      constexpr auto AccessMode = sycl::access_mode::read;
+
+      SECTION((
+          make_section_name(AccessMode, AccessMode, "from 'T' to 'const T'"))) {
+        using SrcDataT = DataT;
+        using DstDataT = const DataT;
+        run_check<SrcDataT, AccessMode, DstDataT, AccessMode>();
+      }
+      SECTION((
+          make_section_name(AccessMode, AccessMode, "from 'const T' to 'T'"))) {
+        using SrcDataT = const DataT;
+        using DstDataT = DataT;
+        run_check<SrcDataT, AccessMode, DstDataT, AccessMode>();
+      }
+    }
+    // From read-write accessor to read-only accessor
+    {
+      constexpr auto SrcAccessMode = sycl::access_mode::read_write;
+      constexpr auto DstAccessMode = sycl::access_mode::read;
+
+      SECTION((make_section_name(SrcAccessMode, DstAccessMode,
+                                 "from 'T' to 'const T'"))) {
+        using SrcDataT = DataT;
+        using DstDataT = const DataT;
+        run_check<SrcDataT, SrcAccessMode, DstDataT, DstAccessMode>();
+      }
+      SECTION((make_section_name(SrcAccessMode, DstAccessMode,
+                                 "from 'const T' to 'T'"))) {
+        using SrcDataT = const DataT;
+        using DstDataT = DataT;
+        run_check<SrcDataT, SrcAccessMode, DstDataT, DstAccessMode>();
+      }
+    }
+  }
+};
+
+/**
+ * @brief Validates implicit conversion for local accessor
+ */
+template <typename DataT, typename DimensionT>
+class check_conversion_local {
+  static constexpr auto AccType = accessor_type::local_accessor;
+  static constexpr int Dimension = DimensionT::value;
+
+  static_assert(!std::is_const_v<DataT>, "No need to pass const type here");
+
+  template <typename SrcDataT, typename DstDataT>
+  inline void run_check() const {
+    // Workarounds to use generic algorithm
+    // TODO: Refactor `check_common_constructor` to make it more generic
+    constexpr auto Target = sycl::target::device;
+    constexpr auto DstAccessMode = std::is_const_v<DstDataT>
+                                       ? sycl::access_mode::read
+                                       : sycl::access_mode::read_write;
+    constexpr int BufferDimension = (Dimension == 0) ? 1 : Dimension;
+    using src_buffer_t = sycl::buffer<SrcDataT, BufferDimension>;
```

Comment:
Can we remove this?

Reply:
Do you ask about `src_buffer_t`? If yes this is an alias for `sycl::buffer<SrcDataT, BufferDimension>` and I suggest do not remove this alias

Reply:
`src_buffer_t` is used in the [accessor_implicit_conversions.h line#221](https://github.com/KhronosGroup/SYCL-CTS/pull/339/files/3494df7ce47e7ed6fb920e794befe8f765aa7e85#diff-df4dec53dbf65f6cc8c5fd5a8957dcaa1be28cd88a1cbb427db5df229b4e790aR221)
This has been done to have the same function's prototype that obtain the same parameters (but some lambdas don't used some parameters)

For example [this](https://github.com/KhronosGroup/SYCL-CTS/pull/339/files/3494df7ce47e7ed6fb920e794befe8f765aa7e85#diff-df4dec53dbf65f6cc8c5fd5a8957dcaa1be28cd88a1cbb427db5df229b4e790aR112-R116) lambda using buffer
But [this](https://github.com/KhronosGroup/SYCL-CTS/pull/339/files/3494df7ce47e7ed6fb920e794befe8f765aa7e85#diff-df4dec53dbf65f6cc8c5fd5a8957dcaa1be28cd88a1cbb427db5df229b4e790aR221-R224) lambda ignoring the buffer